src/harness/testcases/WINNF_FT_S_PPR_testcase.py

```diff
@@ -0,0 +1,391 @@
+    # Moving device_4 to a location within PPA zone
+    device_4['installationParam']['latitude'] = 38.79583
+    device_4['installationParam']['longitude'] = -97.25303
+
+    # Load Grant requests
+    grant_request_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_request_2 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_1.json')))
+    grant_request_3 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_2.json')))
+    grant_request_4 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+
+    # device_b and device_d are Category B
+    # Load Conditional Data
+    self.assertEqual(device_2['cbsdCategory'], 'B')
+    conditionals_device_2 = {
+        'cbsdCategory': device_2['cbsdCategory'],
+        'fccId': device_2['fccId'],
+        'cbsdSerialNumber': device_2['cbsdSerialNumber'],
+        'airInterface': device_2['airInterface'],
+        'installationParam': device_2['installationParam'],
+        'measCapability': device_2['measCapability']
+    }
+
+    self.assertEqual(device_4['cbsdCategory'], 'B')
+    conditionals_device_4 = {
+        'cbsdCategory': device_4['cbsdCategory'],
+        'fccId': device_4['fccId'],
+        'cbsdSerialNumber': device_4['cbsdSerialNumber'],
+        'airInterface': device_4['airInterface'],
+        'installationParam': device_4['installationParam'],
+        'measCapability': device_4['measCapability']
+    }
+
+    conditionals = [conditionals_device_2, conditionals_device_4]
+
+    # Remove conditionals from registration
+    del device_2['cbsdCategory']
+    del device_2['airInterface']
+    del device_2['installationParam']
+    del device_2['measCapability']
+    del device_4['cbsdCategory']
+    del device_4['airInterface']
+    del device_4['installationParam']
+    del device_4['measCapability']
+
+    # Load PPA record
+    ppa_record = json.load(
+      open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+    pal_record = json.load(
+      open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+
+    pal_low_frequency = 3550000000
+    pal_high_frequency = 3570000000
+
+    ppa_record_1, pal_records_1 = makePpaAndPalRecordsConsistent(
+                                  ppa_record,
+                                  [pal_record],
+                                  pal_low_frequency,
+                                  pal_high_frequency,
+                                  'test_user_1'
+    )
+
+    # Registration and grant records
+    cbsd_records_domain_proxy_0 = {
+        'registrationRequests': [device_1, device_2],
+        'grantRequests': [grant_request_1, grant_request_2]
+    }
+    cbsd_records_domain_proxy_1 = {
+        'registrationRequests': [device_3],
+        'grantRequests': [grant_request_3]
+    }
+
+    # Protected entity record
+    protected_entities = {
+        'palRecords': pal_records_1,
+        'ppaRecords': [ppa_record_1]
```

Comment:
The spec says 'A PPA is injected in the SAS before...'.

Need to be just ppa_record_1 (not a list) and the field name would also be singular 'ppaRecord'. 

(Same in PPR.2)

Reply:
Modified   ppaRecord to be singular